### DISCUSSION TITLE
Just a little review! (Plus some wanted features)

### DISCUSSION THREAD

**Discussion by LoneStarz** [keiwando/toonsquid#16] [Ideas]
This is the best animation app I’ve tried on iPad!. Coming from FlipaClip to this app felt like a huge upgrade because it has most of the features I was looking for in an animation app! I love the key frames it is so useful! And the custom brushes are amazing! The outcome feels more detailed as if I were using procreate! So far I’ve only used the app for one day but it already feels amazing! It’s so great to see this app developed from a single person do so much! I found it very easy to learn and got the hang of it in an hour!.

So I haven’t tried all the features, so if these are already in the app please teach me where they are. But if they’re not these features would be greatly appreciated!

• A distort/mesh transform tool: sometimes I like using these to fix something that’s not that far from being perfect, like moving an eye sightly or if something like the mouth needs to be shaped a little. I feel like this tool would be useful if you don’t want to erase and redraw something, a liquify tool would also be a great replacement to this

•reference layers: so I like using the bucket tool to fill in my colors but I don’t like putting it in the same layer as my lineart because I use blending layers to fix the colors and sometimes it messes with my lineart. Something like this or an option to use all the drawing layers in an animation layers as a reference.

•color options: I would like to see this feature in the future because sometimes you want to fix the colors in a single layer and it only needs a little bit of modification.

anyways I’m not like a professional animator or anything but these are the features I would like to see in this app, it would literally be the perfect animation app. Again this app is really good and I recommend it! The best iPad animation app I’ve tried!

**Comment by LoneStarz** [comment]
Forgot to mention: a copy/paste option for multiple frames in the timeline

**Comment by keiwando** [comment]
Thank you so much!

Perspective and mesh transform options are already planned. I have a few things I need to do before starting any work on those but hopefully it won't be too long until these features are ready.

Being able to fill layers based on line work on other layers is also already on the list. I'll bump its priority!

And copying and pasting multiple drawings is currently only possible if all of them are in the same animation layer. I've noted down the request for that to be possible across multiple layers as well.

I'm not quite sure what exactly you mean by "color options". Do you mean things like brightness/saturation etc. adjustments?

**Reply by LoneStarz** [reply]
Yes, I meant the ability to adjust brightness saturation and hue!

**Reply by keiwando** [reply]
Got it, thanks for the clarification! 👍

**Comment by 6xolotl** [comment]
A liquify tool and a reference window would be super useful. I know I’ve come across the need for a liquify tool quite a bit while making my animatics.

As for for the reference window, I feel like these would be useful features if we ever got one:
- Being able to colourpick from the images we load into it. That way, we could put up a character’s ref sheet or a colour palette for the character/scene we’re drawing and be able to keep everything in order. (I still don’t know if this is a common feature with reference windows or if I’ve just gotten lucky with the art programs I use.)
- The Library getting a new section where anything you upload into the reference window will be stored, and can later be reloaded into the window. That way, you won’t have to spend a billion years searching through your photo gallery for that one picture every time you need to reference it again.

**Reply by trinhpxl** [reply]
wdym reference window ? :0 we technically already have one, you just insert an image thru the + button and plop it outside of the canvas
also i think the library already stores ur images/refs in the images tab. iirc- we can also colorpick from the ref images u load in lmao 
![IMG_0258](https://github.com/keiwando/toonsquid/assets/115050959/c2d0c4be-50c9-4039-a157-2f838d4b2432)
lmk if u got any questions ! if u need to elaborate more, feel free to, cuz im confused xD

**Reply by 6xolotl** [reply]
Dropping images in on their own layer works great as is, but I was thinking of it as a quality-of-life thing, y’know? :3

I’d be good without a reference window, but I _really_ hope we get a liquify tool in the future, cuz I’ve needed it a lot ^^;
